### PR TITLE
Do not reset BE DAI pipeline during stop/suspend

### PR DESCRIPTION
**Pull request by ranj063**

### CODE REVIEW

File: sound/soc/sof/intel/hda-dai-ops.c [hunk 1]
```diff
@@ -120,6 +120,27 @@ static struct hdac_ext_stream *hda_get_hext_stream(struct snd_sof_dev *sdev,
 	return snd_soc_dai_get_dma_data(cpu_dai, substream);
 }
 
+static struct hdac_ext_stream *hda_ipc4_get_hext_stream(struct snd_sof_dev *sdev,
```
Comment (plbossart) [inline sound/soc/sof/intel/hda-dai-ops.c:123]: Commit message. Maybe we can reword the end:

With this change, we can remove the HDaudio-specific setting in sof_ipc4_dai_config(), which will further simplify support for DMIC/SSP/SoundWire in LNL.

File: sound/soc/sof/intel/hda-dai-ops.c [hunk 2]
```diff
@@ -120,6 +120,27 @@ static struct hdac_ext_stream *hda_get_hext_stream(struct snd_sof_dev *sdev,
...
+	struct snd_soc_dapm_widget *w;
+
+	w = snd_soc_dai_get_widget(cpu_dai, substream->stream);
+	swidget = w->dobj.private;
+	pipe_widget = swidget->spipe->pipe_widget;
+	pipeline = pipe_widget->private;
+
+	/* mark pipeline so that it can be skipped during FE trigger */
+	if (!pipeline->skip_during_fe_trigger)
+		pipeline->skip_during_fe_trigger = true;
```
Comment (plbossart) [inline sound/soc/sof/intel/hda-dai-ops.c]: do this unconditionally, there's no real benefit to have a test here?

File: sound/soc/sof/intel/hda-dai-ops.c [hunk 3]
```diff
@@ -312,15 +312,6 @@ static int hda_ipc4_post_trigger(struct snd_sof_dev *sdev, struct snd_soc_dai *c
...
 	case SNDRV_PCM_TRIGGER_STOP:
-	{
-		ret = sof_ipc4_set_pipeline_state(sdev, pipe_widget->instance_id,
-						  SOF_IPC4_PIPE_RESET);
-		if (ret < 0)
-			return ret;
-
-		pipeline->state = SOF_IPC4_PIPE_RESET;
-		break;
-	}
```
Comment (plbossart) [inline sound/soc/sof/intel/hda-dai-ops.c:302]: commit message:
duplicated "So so"

also we need to explain that this follows the simplification already done for FEs, and that the DAI trigger only change the pipeline states between PAUSED and RUNNING.

**Comment by ranj063** [comment]
SOFCI TEST

File: sound/soc/sof/intel/hda-dai-ops.c [hunk 4]
```diff
@@ -222,6 +222,7 @@ static struct hdac_ext_link *hda_get_hlink(struct snd_sof_dev *sdev,
 static int hda_ipc4_pre_trigger(struct snd_sof_dev *sdev, struct snd_soc_dai *cpu_dai,
 				struct snd_pcm_substream *substream, int cmd)
 {
+	struct sof_ipc4_fw_data *ipc4_data = sdev->private;
```
Comment (plbossart) [inline sound/soc/sof/intel/hda-dai-ops.c:225]: commit message: PASUED typo

**Review by plbossart — CHANGES_REQUESTED** [review]

**Comment by ranj063** [comment]
SOFCI TEST

**Review by plbossart — APPROVED** [review]

File: sound/soc/sof/intel/hda-dai-ops.c [hunk 5]
```diff
@@ -311,7 +331,7 @@ static int hda_ipc4_post_trigger(struct snd_sof_dev *sdev, struct snd_soc_dai *c
 }
 
 static const struct hda_dai_widget_dma_ops hda_ipc4_dma_ops = {
-	.get_hext_stream = hda_get_hext_stream,
```
Comment (plbossart) [inline sound/soc/sof/intel/hda-dai-ops.c:314]: actually question @ranj063: is this intentional that we don't modify the chain DMA ?

````c
static const struct hda_dai_widget_dma_ops hda_ipc4_chain_dma_ops = {
	.get_hext_stream = hda_get_hext_stream,
````

Comment (ranj063) [reply sound/soc/sof/intel/hda-dai-ops.c:314]: @plbossart yes. Chain DMA does not use the pipelines in topology

**Comment by ranj063** [comment]
SOFCI TEST

**Comment by fredoh9** [comment]
SOFCI TEST

**Comment by fredoh9** [comment]
Device test results was inconsistent, especially on ADLP_RVP_NOCODEC_IPC4ZPH. I suspect device specific issue. Last test results are all green.

**Review by bardliao — APPROVED** [review]

**Comment by plbossart** [comment]
@ujfalusi are you good with this PR? we need to merge it soon, it gates other PRs

**Pull request merged**